Marlin/src/feature/fwretract.cpp

```diff
@@ -145,34 +141,40 @@ void FWRetract::retract(const bool retracting
     prepare_move_to_destination();
 
     // Is a Z hop set, and has the hop not yet been done?
-    if (has_zhop) {
-      hop_amount += retract_zlift;                // Carriage is raised for retraction hop
+    // No double zlifting 
+    // Feedrate to the max
+    if (has_zhop && !hop_amount) {
+      hop_amount += retract_zlift;  	  // Carriage is raised for retraction hop
+      float temp_feedrate_mm_s=feedrate_mm_s; // backup the current feedrate 
+      feedrate_mm_s = planner.max_feedrate_mm_s[Z_AXIS]; // Z feedrate to max
       current_position[Z_AXIS] -= retract_zlift;  // Pretend current pos is lower. Next move raises Z.
       SYNC_PLAN_POSITION_KINEMATIC();             // Set the planner to the new position
       prepare_move_to_destination();              // Raise up to the old current pos
+      feedrate_mm_s = temp_feedrate_mm_s  ; //feedrate restoration
     }
   }
   else {
     // If a hop was done and Z hasn't changed, undo the Z hop
-    if (hop_amount && NEAR(hop_height, destination[Z_AXIS])) {
-      current_position[Z_AXIS] += hop_amount;     // Pretend current pos is higher. Next move lowers Z.
```

Comment:
Because of this line , simplify3d is down , because moving Z , between retractation
If the machine makes a hop , it can recover
A printer never go down , only a Geek with modified Gcode files can make a reversed Z movement , and only at the starting just to see the printer going everywhere up and down for nothing 😄